### PR TITLE
feat: Report enduser.id with Session Replay

### PR DESCRIPTION
**Pull request by metal-messiah**
Report enduser.id if set with Session Replay data to improve faceting the Session Replay UX
---

### Overview
This PR adds the enduser id to outgoing SR payloads to help facilitate increased faceting in Session Replay UXs.

### Related Issue(s)
https://new-relic.atlassian.net/browse/NR-180218

### CODE REVIEW

**Review by patrickhousley — APPROVED** [review]
Link to spec update PR pls

**Comment by metal-messiah** [comment]
https://source.datanerd.us/agents/agent-specs/blob/main/browser/Blob-Request.md#session-replay

File: src/features/session_replay/aggregate/index.js [hunk 1]
```diff
@@ -67,6 +67,8 @@ let recorder, gzipper, u8
 export const MAX_PAYLOAD_SIZE = 1000000
 /** Unloading caps around 64kb */
 export const IDEAL_PAYLOAD_SIZE = 64000
+/** Reserved room for query param attrs */
+const QUERY_PARAM_PADDING = 5000
```
Comment (patrickhousley) [inline src/features/session_replay/aggregate/index.js:71]: Just curious, how much room does this leave us in the required params section?

**Review by patrickhousley — APPROVED** [review]

File: src/features/session_replay/aggregate/index.js (hunk 1 above)
Comment (patrickhousley) [reply src/features/session_replay/aggregate/index.js:71]: Yea I guess what I am asking is how much of the 5kb are we currently consuming with the required attributes.

Comment (metal-messiah) [reply src/features/session_replay/aggregate/index.js:71]: right now the req'd params encompass about 0.5kb

**Pull request merged**